### DISCUSSION TITLE
Netxjs Middleware with localstorage or cookies

### DISCUSSION THREAD

**Discussion by Lalman888** [vercel/next.js#52463] [Help]
### Summary

I want to ask how can I use to protect route using localStorage if possible or cookies . how to set token in cookies . I am using login api and register api for auth

**Comment by hyjoong** [comment]
Since middleware runs on the server-side, you cannot use browser-specific global objects like window or localStorage.

If you want to store the token in a cookie when a user logs in and apply route guards in the middleware, you can use the following code snippet. However, please note that the code below only checks for the presence of the refresh token. You would need to implement it according to your specific requirements.

```typescript
export async function middleware(req: NextRequest) {
  const token = req.headers.get('token') // get token from request header 

  if (!token) {
    const signinUrl = new URL('/signin', req.url)
    return NextResponse.redirect(signinUrl)
  }

  return NextResponse.next()
}

// Here you can specify all the paths for which this middleware function should run
// Supports both a single string value or an array of matchers
export const config = {
  matcher: ['/api/auth/:path*'],
}
```

https://shipsaas.com/blog/create-protected-route-nextjs
